### DISCUSSION TITLE
通过 s cli 访问组件时是否真的有必要下载组件的源码到本地进行编译?

### DISCUSSION THREAD

**Discussion by DevDengChao** [Serverless-Devs/Serverless-Devs#153] [General]
在调用了几个不同的组件后, 我发现 `~/components` 文件夹占用了几百兆的硬盘空间, 让本就不充裕的 C 盘变得更加雪上加霜.

调查了一番后发现, `s` 将组件的源代码库 (特定版本的快照) 下载到了本地, 并且在各个组件对应的文件夹内进行了编译.

组件的源代码快照 :
![组件的源代码快照](https://user-images.githubusercontent.com/16363180/126999669-445453ac-d9ee-4ef2-86c4-8308391d934c.png)

组件文件夹 :
![组件文件夹](https://user-images.githubusercontent.com/16363180/126999670-f852fc0c-ac5b-4dba-884a-374e0fd904ce.png)

因为 `s cli` 在后续使用的过程中并不需要 `dist` 文件夹以外的内容, 所以我认为在客户端本地进行编译只是白白浪费公网流量和硬盘容量.

**Comment by anycodes** [comment]
我们接下来会通过几个方面来解决这个事情：
1. 我们所有的代码采用ncc编译，去掉所有无用的文件；
2. 我们默认不加版本号的下载，可以确保每个组件占用最低空间；

**Reply by heimanba** [reply]
cli版本 在 2.0.84 之后，以上能力均已具备，
